### ISSUE TITLE
Suggestion: StructuredOutputConverter implementations could have configured amount of retries when convert() is failing to parse response

### ISSUE THREAD

**Issue by Sameperson** [spring-projects/spring-ai#947]
**Current Behavior**

Currently, existing convert() implementations (BeanOutputConverter, ListOutputConverter, MapOutputConverter) throw exceptions when the response can't be parsed.

**Proposed Behavior**

I saw Spring AI has retry properties: https://docs.spring.io/spring-ai/reference/api/chat/openai-chat.html#_chat_properties
But these are for the initial calls to the model if I understood correctly, not the converters. 

Would it be a good idea to set the optional amount of retries for the converters too? If parsing fails, we can retry to call the AI model in hopes of getting a better response.

**Context**

I am getting the wrong output from the model from time to time, so I configured retries in my code. But what do you think about having this feature in Spring AI?

**Comment by ThomasVitale** [comment]
I like the suggestion. LangChain provides a couple of OutputParsers for dealing with errors:

* an `[OutputFixingParser](https://python.langchain.com/v0.2/docs/how_to/output_parser_fixing/)` which contacts a second model in case the default one fails in producing a valid output;
* a [`RetryOutputParser`](https://python.langchain.com/v0.2/docs/how_to/output_parser_retry/) which tries again with the same model with the hope to get a better response.

It would be nice to have something similar for the `StructuredOutputConverter` infrastructure in Spring AI:

**Comment by etlweather** [comment]
... yes, I am going to try the `ListOutputConverter` - that might work better in my case... still would like the retry!

**Comment by topone2024** [comment]
I agree with your suggestion that configuring the retry count for StructuredOutputConverter implementations when the convert() method fails to parse the response could significantly reduce the workload caused by parsing failures. This would improve the system's robustness and user experience. I look forward to Spring AI implementing this feature soon.
